kv/test_local_server.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Spencer Kimball (spencer.kimball@gmail.com)
+
+package kv
```

Comment:
This actually presents a difficulty - if you move a context like this into a test package, and then you can't use it in any package containing an object it creates. This means you can't use it in tests which access unexported members of those packages.  

For example, most of this is VERY similar to the code in `storage.testContext`, and I tried to extract into a test package in my first week; however, that class is used in quite a few unit tests which access unexported objects in the `storage` package, which wouldn't work if it was in a separate package.  I didn't feel good about exporting `storage.TestContext` to use it in other packages;  However, I also didn't feel good about duplicating it into a test package for the purpose of export. That's why I created my own unexported implementation in the `ts` package, because I was obsessing over the correct solution.

I actually don't know what the correct solution is; exporting test code like this directly from code packages allows everything to work, but it also seems "unclean" somehow. I think the "goism" here would be to duplicate the code, but I guess I'm still not completely comfortable with that aspect of go.